airflow-core/src/airflow/utils/warnings.py

```diff
@@ -14,27 +14,9 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-from __future__ import annotations
 
-import warnings
-from collections.abc import Generator
-from contextlib import contextmanager
+from __future__ import annotations
 
+from airflow.utils import _deprecate_this_module
 
-@contextmanager
-def capture_with_reraise() -> Generator[list[warnings.WarningMessage], None, None]:
```

Comment:
is this part of the public API? If not we can just remove it

Reply:
We have no rules on if things in `airflow.utils` are public. A lot of other modules in that package are used by users, and we never said they shouldn’t (and probably can’t at this point).